pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -218,6 +219,18 @@ func (ctrl *VMSnapshotController) updateVMSnapshotContent(content *snapshotv1.Vi
 				continue
 			}
 
+			if !checkedGuestAgentParticipation {
+				vmSnapshot, err := ctrl.getVMSnapshot(content)
+				if err != nil {
+					return 0, err
+				}
+				err = ctrl.freezeGuestFSIfNeeded(vmSnapshot)
```

Comment:
I am not sure, supposedly the call is immediate, sends the command which should happen and return, I'm not sure if with busy fs it might take more time. I don't think we should call the freeze, continue with other snapshots and return to this snapshot when it completes, but i'm not sure.. how can we check? what do you think is best?

Reply:
I guess let's not get too complicated now.  I think it would be wise to increase snapshot controller threads here:  https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/application.go#L430.  Maybe double?

Should also think about a way to log/track how long that call is taking.

Reply:
ok sure

Reply:
@mhenriks I fixed all the other comments and add two commits one for functional tests and another for the increase of the number of the threads and added logging of time for the freeze/unfreeze, from what I saw the freeze unfreeze of vms in the tests took around 100ms to freeze and 25ms to unfreeze